lisa/sut_orchestrator/azure/common.py

```diff
@@ -1232,36 +1232,49 @@ def generate_sas_token(
 
 
 def get_blob_service_client(
-    credential: Any,
-    subscription_id: str,
-    cloud: Cloud,
-    account_name: str,
-    resource_group_name: str,
+    credential: Optional[Any] = None,
+    subscription_id: Optional[str] = None,
+    cloud: Optional[Cloud] = None,
+    account_name: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+    connection_string: Optional[str] = None,
 ) -> BlobServiceClient:
     """
     Create a Azure Storage container if it does not exist.
     """
-    shared_key_credential = get_storage_credential(
-        credential=credential,
-        subscription_id=subscription_id,
-        cloud=cloud,
-        account_name=account_name,
-        resource_group_name=resource_group_name,
-    )
-    blob_service_client = BlobServiceClient(
-        f"https://{account_name}.blob.{cloud.suffixes.storage_endpoint}",
-        shared_key_credential,
-    )
+    blob_service_client: BlobServiceClient
+    if connection_string:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+    else:
+        assert subscription_id, "subscription_id is required."
+        assert cloud, "cloud is required."
```

Comment:
We should give the public cloud as default.

Reply:
updated